client/src/components/Nav/SettingsTabs/Data/ExportConversations.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import { ArrowRightFromLine } from 'lucide-react';
+import { useExportConversationsMutation } from '~/data-provider';
+import { useLocalize } from '~/hooks';
+import { useToastContext } from '~/Providers';
+
+function ExportConversations() {
+  const localize = useLocalize();
+  const { showToast } = useToastContext();
+  const [errors, setErrors] = useState<string[]>([]);
+  const setError = (error: string) => setErrors((prevErrors) => [...prevErrors, error]);
+
+  const exportFile = useExportConversationsMutation({
+    onSuccess: (data) => {
+      console.log('Exporting started', data);
+
+      showToast({ message: localize('com_ui_export_conversation_success') });
+
+      // Directly initiate download here if `data` contains downloadable content or URL
+      downloadConversationsJsonFile(data);
+    },
+    onError: (error) => {
+      console.error('Error: ', error);
+      setError(
+        (error as { response: { data: { message?: string } } }).response.data.message ??
+          'An error occurred while exporting the file.',
+      );
+      showToast({ message: localize('com_ui_export_conversation_error'), status: 'error' });
+    },
+  });
+
+  const startExport = () => {
+    const formdata = new FormData();
+    exportFile.mutate(formdata);
+  };
+
+  const downloadConversationsJsonFile = (data) => {
+    // Assuming `data` is the downloadable content; adjust as necessary for your use case
+    const blob = new Blob([JSON.stringify(data, null, 2)], { type: 'application/json' });
```

Comment:
Hm, this seems not optimal. We kind of downloading potentially big file then storing it in memory. Can't we download it directly?